src/cclib/parser/orcaparser.py

```diff
@@ -643,23 +643,144 @@ def extract(self, inputfile, line):
                 self.etsecs.append(sec)
                 line = next(inputfile)
 
-        # This will parse etoscs for TD calculations, but note that ORCA generally
-        # prints two sets, one based on the length form of transition dipole moments,
-        # the other based on the velocity form. Although these should be identical
-        # in the basis set limit, in practice they are rarely the same. Here we will
-        # effectively parse just the spectrum based on the length-form.
-        if (line[25:44] == "ABSORPTION SPECTRUM" or line[9:28] == "ABSORPTION SPECTRUM") and not hasattr(self, "etoscs"):
-
-            self.skip_lines(inputfile, ['d', 'header', 'header', 'd'])
-
-            self.etoscs = []
-            for x in self.etsyms:
-                osc = next(inputfile).split()[3]
-                if osc == "spin":  # "spin forbidden"
-                    osc = 0
-                else:
-                    osc = float(osc)
-                self.etoscs.append(osc)
+
+        # Parse the various absorption spectra for TDDFT and ROCIS
+        if 'ABSORPTION SPECTRUM' in line or 'ELECTRIC DIPOLE' in line:
```

Comment:
Would anything fail if this was changed to only `if 'ABSORPTION SPECTRUM' in line:`?